# Data sources

**Dataset 1: University GPA/ACT/SAT Acceptance Info (2020)**

URL: https://www.kaggle.com/datasets/tacotown414/university-gpaactsat-acceptance-2020


**Dataset 2: SAT to GPA**

URL: https://www.kaggle.com/datasets/farhansadeek/sat-to-gpa

**Dataset3: University Admission Dataset**

URL: https://www.kaggle.com/datasets/farhansadeek/university-admission-dataset

# Deliverable 2

## Importing Libraries

In [386]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Parsing Datasets

Firstly, all the datasets - `.csv` files will be read using two methods:
-  The `open()` function with the file name, `'r'` being passed in as parameters, and then use the `readlines()` function to read the parsed csv file
-  Use the `pd.read_csv()` function which returns the dataframe version of the dataset

In [387]:
with open('University Acceptance Info.csv', 'r') as file:
  dataset1 = file.readlines()

#create a new list acceptance_info to append the cleaned data from the .csv file
  acceptance_info = list()

  for d in dataset1:
    acceptance_info.append(d.strip('\n').split(',')) #eliminating the '\n' from dataset1's objects 

In [388]:
with open('SAT to GPA.csv', 'r') as file:
  dataset2 = file.readlines()

In [389]:
score_importance = pd.read_csv('Admission Data.csv')

## Dataset 1: University GPA/SAT Acceptance Info

- This dataset shows the average standardized tests scores, as well as GPAs corresponding to the best 295 universities in the country. Also, the acceptance rates and financial statistics are included to demonstrate how rigorous are the universities. 

- Considering the average SAT, average GPA and the universities' rankings helps determine whether the SAT scores or GPA of top-tier universities' admitted students match their school's reputation more 

-  We can spot if the students' GPA or SAT is more average than the 'expected' score to get in those universities to figure the scores' weight in this situation  

### Step 1: Construct a Dictionary of Lists (DoL) representation 

Function `uni_score_average()` takes L (we can pass in a parsed dataset) as one argument and then returns it as a Dictionary of Lists (DoL)

In [390]:
def uni_score_average(L):
  DoL = dict() #creating an empty dictionary to update the keys-values to construct a DoL

  for i in range(len(L[0])): #selecting the first key of the dictionary as the index 
    DoL[L[0][i]] = list() #for each key of the DoL, create a list of corresponding values
    
    for j in range(1, len(L)):
      dict_value = L[j][i] #with a loop going through every key's list of values, assign them to a variable called dict_value
      DoL[L[0][i]].append(dict_value) #append the values to every list of values within each key
  
  return DoL

uni_score_average(acceptance_info)

{'\ufeffKEYid': ['2627',
  '2155',
  '1774',
  '1426',
  '2707',
  '2178',
  '1305',
  '3378',
  '2920',
  '1131',
  '2573',
  '2077',
  '1739',
  '3401',
  '2711',
  '3604',
  '3535',
  '1840',
  '2520',
  '1445',
  '1564',
  '1312',
  '1315',
  '1328',
  '3242',
  '6968',
  '2978',
  '9092',
  '2219',
  '2785',
  '2974',
  '2128',
  '3705',
  '2133',
  '1569',
  '2894',
  '2130',
  '3024',
  '1320',
  '2199',
  '2029',
  '2803',
  '1314',
  '1317',
  '1535',
  '3289',
  '1264',
  '1313',
  '1536',
  '3895',
  '3388',
  '6965',
  '1775',
  '6883',
  '1598',
  '1444',
  '1825',
  '29013',
  '3658',
  '3798',
  '3670',
  '2722',
  '3613',
  '2882',
  '2103',
  '2233',
  '3425',
  '3379',
  '1434',
  '6964',
  '2639',
  '10366',
  '3969',
  '3754',
  '6967',
  '1348',
  '2221',
  '7104',
  '3636',
  '1892',
  '2139',
  '1489',
  '2290',
  '2972',
  '1321',
  '1431',
  '2836',
  '1371',
  '3185',
  '1809',
  '3863',
  '1370',
  '10395',
  '3256',
  '2506',
  '2903',
  '2806',
  '2838',
  

### Step 2: Construct a pandas DataFrame from the DoL created

In [391]:
acceptance_df = pd.DataFrame(uni_score_average(acceptance_info))
acceptance_df.head(60)

KEYid                                     School Name State  \
0    2627                            Princeton University    NJ   
1    2155                              Harvard University    MA   
2    1774                           University of Chicago    IL   
3    1426                                 Yale University    CT   
4    2707                             Columbia University    NY   
5    2178           Massachusetts Institute of Technology    MA   
6    1305                             Stanford University    CA   
7    3378                      University of Pennsylvania    PA   
8    2920                                 Duke University    NC   
9    1131              California Institute of Technology    CA   
10   2573                               Dartmouth College    NH   
11   2077                        Johns Hopkins University    MD   
12   1739                         Northwestern University    IL   
13   3401                                Brown University    RI   
14   2711                              Cornell University    NY   
15   3604                                 Rice University    TX   
16   3535                           Vanderbilt University    TN   
17   1840                        University of Notre Dame    IN   
18   2520              Washington University in St. Louis    MO   
19   1445                           Georgetown University    DC   
20   1564                                Emory University    GA   
21   1312              University of California--Berkeley    CA   
22   1315           University of California--Los Angeles    CA   
23   1328               University of Southern California    CA   
24   3242                      Carnegie Mellon University    PA   
25   6968                          University of Virginia    VA   
26   2978                          Wake Forest University    NC   
27   9092               University of Michigan--Ann Arbor    MI   
28   2219                                Tufts University    MA   
29   2785                             New York University    NY   
30   2974       University of North Carolina--Chapel Hill    NC   
31   2128                                  Boston College    MA   
32   3705                     College of William and Mary    VA   
33   2133                             Brandeis University    MA   
34   1569                 Georgia Institute of Technology    GA   
35   2894                         University of Rochester    NY   
36   2130                               Boston University    MA   
37   3024                 Case Western Reserve University    OH   
38   1320         University of California--Santa Barbara    CA   
39   2199                         Northeastern University    MA   
40   2029                               Tulane University    LA   
41   2803                Rensselaer Polytechnic Institute    NY   
42   1314                University of California--Irvine    CA   
43   1317             University of California--San Diego    CA   
44   1535                           University of Florida    FL   
45   3289                               Lehigh University    PA   
46   1264                           Pepperdine University    CA   
47   1313                 University of California--Davis    CA   
48   1536                             University of Miami    FL   
49   3895                University of Wisconsin--Madison    WI   
50   3388                            Villanova University    PA   
51   6965  Pennsylvania State University--University Park    PA   
52   1775        University of Illinois--Urbana-Champaign    IL   
53   6883                 Ohio State University--Columbus    OH   
54   1598                           University of Georgia    GA   
55   1444                    George Washington University    DC   
56   1825               Purdue University--West Lafayette    IN   
57  29013                       University of Connecticut    CT   
58   3658                     University of Texas-

## Dataset 2: Individual SAT and GPA scores comparison

-  This dataset represents the connection between the SAT scores and GPAs of each of 180 students surveyed
-  By comparing each student's SAT and GPA, we can clarify whether students with high GPA will always have high SAT scores or vice versa

### Step 1: Construct a List of Lists (LoL) representation

Function `is_highGPA_highSAT()` takes one argument `data` (we can pass in a parsed dataset) and returns it as a list of lists

In [392]:
def is_highGPA_highSAT(data):

  SATtoGPA = list()

  for e in data: #iterate over all lines

    #strip the objects from whitespace using strip() function

    SATtoGPA.append(e.strip('\n').split(',')) #split the objects on the delimeter ',' using split() function
    
  return SATtoGPA
is_highGPA_highSAT(dataset2)  

[['SAT Score', 'GPA'],
 ['1600', '4'],
 ['1593', '4'],
 ['1587', '4'],
 ['1580', '4'],
 ['1573', '4'],
 ['1567', '4'],
 ['1560', '4'],
 ['1553', '4'],
 ['1547', '4'],
 ['1540', '4'],
 ['1533', '4'],
 ['1527', '4'],
 ['1520', '3.99'],
 ['1513', '3.99'],
 ['1507', '3.99'],
 ['1500', '3.99'],
 ['1493', '3.99'],
 ['1487', '3.99'],
 ['1480', '3.99'],
 ['1473', '3.99'],
 ['1467', '3.99'],
 ['1460', '3.99'],
 ['1453', '3.99'],
 ['1447', '3.99'],
 ['1440', '3.99'],
 ['1433', '3.99'],
 ['1427', '3.99'],
 ['1420', '3.99'],
 ['1413', '3.99'],
 ['1407', '3.99'],
 ['1400', '3.99'],
 ['1393', '3.99'],
 ['1387', '3.99'],
 ['1380', '3.99'],
 ['1373', '3.98'],
 ['1367', '3.98'],
 ['1360', '3.98'],
 ['1353', '3.98'],
 ['1347', '3.98'],
 ['1340', '3.98'],
 ['1333', '3.98'],
 ['1327', '3.98'],
 ['1320', '3.97'],
 ['1313', '3.97'],
 ['1307', '3.97'],
 ['1300', '3.97'],
 ['1293', '3.97'],
 ['1287', '3.96'],
 ['1280', '3.96'],
 ['1273', '3.96'],
 ['1267', '3.95'],
 ['1260', '3.95'],
 ['1253', '3.95'],
 ['124

### Step 2: Construct a pandas DataFrame from the LoL created

In [393]:
SATvsGPA = pd.DataFrame(is_highGPA_highSAT(dataset2))
SATvsGPA

0    1
0    SAT Score  GPA
1         1600    4
2         1593    4
3         1587    4
4         1580    4
..         ...  ...
177        427    0
178        420    0
179        413    0
180        407    0
181        400    0

[182 rows x 2 columns]

## Dataset 3: University Admission Dataset

-  This dataset will be directly converted to a pandas DataFrame

-  Dataset 3 illustrates how different universities rank admission factors such as GPA, standardized tests, academic rigor, essays, recommendations, etc with four levels of consideration:
    +  0 = Not Considered
    +  1 = Considered
    +  2 = Important
    +  3 = Very Important 

-  From the dataset, we can determine if universities weigh test scores more than GPAs or vice versa, which indicates SAT or GPA will get students admitted to top institutions in the US


Function `score_importance()` has one argument `file_name`. When the function is called, we pass along the csv file `Admission Data.csv`, and it will return a DataFrame 

In [394]:
def score_importance(file_name):
  dataframe = pd.read_csv(file_name) #read the csv file and convert it to a dataframe simultaneously
  return dataframe
  
score_importance('Admission Data.csv')

Name of University  Rigor  Rank  GPA  \
0                           Princeton University      3     3    3   
1                               Brown University      3     3    3   
2   California Institute of Technology (Caltech)      3     2    2   
3                             Cornell University      3     2    3   
4                              Dartmouth College      3     3    3   
5                             Harvard University      1     1    1   
6    Massachusetts Institute of Technology (MIT)      2     1    2   
7                            Stanford University      3     3    3   
8                     University of Pennsylvania      3     2    1   
9     Washington University in St. Louis (WashU)      3     3    3   
10                               Yale University      3     3    3   
11                               Rice University      3     3    3   
12                               Duke University      3     3    3   
13                         Vanderbilt University      3     3    3   
14                       Northwestern University      3     3    3   
15       University of Southern California (USC)      3     0    3   
16                      University of Notre Dame      3     2    2   
17              Carnegie Mellon University (CMU)      3     3    3   
18                         Georgetown University      3     3    3   
19                     New York University (NYU)      3     3    3   
20               The Ohio State University (OSU)      3     3    3   
21                              Miami University      3     3    3   
22         Case Western Reserve University (CWR)      3     3    3   

    Test Score  Essay  Recommendation  Interview  Extracurricular  Talent  \
0            3      3               3          1                3       3   
1            3      3               3          1                2       3   
2            3      3               3          0                2       1   
3            3      3               3          1                3       3   
4            3      3               3          1                3       2   
5            1      1               1          1                1       1   
6            2      2               2          2                2       2   
7            3      3               3          1                3       3   
8            1      1               1          2                2       2   
9            3      3               3          1                2       3   
10           1      3               3          1                3       3   
11           3      3               3          1                3       3   
12           3      3               3          1                3       3   
13           3      3               2          1                3       1   
14           1      2               2          1                2       2   
15           3      3               3          0                2       2   
16           1      2               2          0                2       2   
17           2      2               2          0                1       2   
18           3      3               3          2                2       3   
19           3      2               2          1                2       3   
20           2      1               1          0                2       2   
21           3      3               3          0                1       3   
22           1      2               2          2                1       2   

    Personality  First Generation  Religious  State Residency  Race  \
0             3                 1        0.0                0     1   
1             3                 1        0.0                1     1   
2             3                 1        0.0                0     1   
3             3                 1        NaN                1     1   
4             3                 0        1.0                0     1   
5             1                 1        0.0                0     1   
6  

# Data Cleaning 

## Cleaning Dataset 1

-  This dataset shows the universities, average test scores, GPAs, etc corresponding to the best 295 universities in the US

-  Since we only need the SAT, GPA 

-  Therefore, in the new dataframe, the independent variables ae SAT Avg, HS GPA Avg, and the dependent variables are Acceptance Rate, School Ranking 

-  Then, we will process the N/A values  

### Step 1: Deleting unnecessary columns/rows

-  We ONLY need the columns: School Name, School Ranking, SAT Avg, HS GPA Avg, Acceptance Rate 

  ----> The unnecessary columns will be eliminated using the `drop()` method  

In [395]:
acceptance_df.drop(['State', 'Percent Receiving Aid', 'City', 'ACT Avg', 'Total Enrollment', 'Cost after Aid', 'Total Tuiton', 'Is Public?'], axis='columns', inplace=True)
acceptance_df.drop(acceptance_df.columns[0], axis=1)
cleaned_acceptance_df = acceptance_df.loc[acceptance_df['School Ranking'] != '-1']

### Step 2: Processing N/A values

-  Since the first dataset's missing values are just blank strings by default, we have to replace them with N/A so that to eliminate all rows containing those values



In [396]:
#Replace all missing values in the DataFrame with N/A 
for i in range(len(cleaned_acceptance_df)):
  if cleaned_acceptance_df.loc[i, 'HS GPA Avg'] == '':
    cleaned_acceptance_df.loc[i, 'HS GPA Avg'] = np.nan

  elif cleaned_acceptance_df.loc[i, 'SAT Avg'] == '':
    cleaned_acceptance_df.loc[i, 'SAT Avg'] = np.nan

  elif cleaned_acceptance_df.loc[i, 'School Ranking'] == '':
    cleaned_acceptance_df.loc[i, 'School Ranking'] = np.nan

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)
/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:723: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value, self.name)


In [397]:
#Check how many NaN values are there in the entire DataFrame
cleaned_acceptance_df.isnull().sum()

KEYid              0
School Name         0
School Ranking      0
SAT Avg             4
HS GPA Avg         44
Acceptance Rate     0
dtype: int64

In [398]:
#eliminating all rows containing N/A values
cleaned_acceptance_df.dropna(inplace=True)
cleaned_acceptance_df.isnull().sum()

/usr/local/lib/python3.7/dist-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


KEYid             0
School Name        0
School Ranking     0
SAT Avg            0
HS GPA Avg         0
Acceptance Rate    0
dtype: int64

After dealing with the N/A values, we have to sort the columns according to both the GPA Avg values and School Rankings to spot top-tier schools that have minor to big difference between HS GPA average and SAT score average 

In [399]:
#sort the dataframe according to HS GPA average

df_GPAsorted = cleaned_acceptance_df.sort_values(by='HS GPA Avg', ascending=False)
df_GPAsorted.head()

KEYid               School Name School Ranking SAT Avg HS GPA Avg  \
1    2155        Harvard University              2    1430          4   
25   6968    University of Virginia             25    1240          4   
66   3425        Clemson University             67    1150          4   
67   3379  University of Pittsburgh             68    1190          4   
54   1598     University of Georgia             54    1140          4   

   Acceptance Rate  
1            5.00%  
25          30.00%  
66          51.00%  
67          55.00%  
54          54.00%

In [400]:
cleaned_acceptance_df.head(30)

KEYid                                     School Name School Ranking  \
0    2627                            Princeton University              1   
1    2155                              Harvard University              2   
2    1774                           University of Chicago              3   
7    3378                      University of Pennsylvania              8   
11   2077                        Johns Hopkins University             11   
16   3535                           Vanderbilt University             14   
20   1564                                Emory University             21   
21   1312              University of California--Berkeley             21   
22   1315           University of California--Los Angeles             21   
23   1328               University of Southern California             21   
24   3242                      Carnegie Mellon University             25   
25   6968                          University of Virginia             25   
27   9092               University of Michigan--Ann Arbor             28   
29   2785                             New York University             30   
30   2974       University of North Carolina--Chapel Hill             30   
33   2133                             Brandeis University             34   
34   1569                 Georgia Institute of Technology             34   
36   2130                               Boston University             37   
38   1320         University of California--Santa Barbara             37   
40   2029                               Tulane University             40   
41   2803                Rensselaer Polytechnic Institute             42   
42   1314                University of California--Irvine             42   
43   1317             University of California--San Diego             42   
44   1535                           University of Florida             42   
46   1264                           Pepperdine University             46   
47   1313                 University of California--Davis             46   
48   1536                             University of Miami             46   
49   3895                University of Wisconsin--Madison             46   
50   3388                            Villanova University             46   
51   6965  Pennsylvania State University--University Park             52   

   SAT Avg HS GPA Avg Acceptance Rate  
0     1400        3.9           7.00%  
1     1430          4           5.00%  
2     1450          4           8.00%  
7     1380        3.9           9.00%  
11    1420        3.9          12.00%  
16    1420        3.8          11.00%  
20    1290        3.7          25.00%  
21    1300        3.9          16.00%  
22    1160        3.9          18.00%  
23    1280        3.7          17.00%  
24    1380        3.8          22.00%  
25    1240          4          30.00%  
27    1310        3.8          29.00%  
29    1250        3.6          32.00%  
30    1190          4          27.00%  
33    1270        3.9          33.00%  
34    1320        3.9          26.00%  
36    1220        3.6          29.00%  
38    1140          4          36.00%  
40    1250        3.5          26.00%  
41    1280        3.9          44.00%  
42    1060          4          41.00%  
43    1140          4          36.00%  
44    1180          4          46.00%  
46    1110        3.6          37.00%  
47    1050          4          42.00%  
48    1210        3.6          38.00%  
49    1234        3.8          53.00%  
50    1220        3.7          44.00%  
51    1090        3.6          56.00%

## Cleaning dataset 2

### Step 1: Name Columns

Since this is a List of Lists (LoL) representation of data, the originally converted DataFrame will not have column names, instead, the 'SAT' and 'GPA' label is the first row of the column. Therefore, we have to get rid of that first row and name the two columns

In [401]:
#Name the columns
SATvsGPA.columns = ['SAT', 'GPA']

#delete the first row of the dataframe since we have already named the columns
SATvsGPA.drop(labels=0, axis=0, inplace=True)
SATvsGPA

SAT GPA
1    1600   4
2    1593   4
3    1587   4
4    1580   4
5    1573   4
..    ...  ..
177   427   0
178   420   0
179   413   0
180   407   0
181   400   0

[181 rows x 2 columns]

### Step 2: Processing N/A values
Check if there is any N/A values using the `isnull()` function and calculate the number of N/A values spotted with `sum()` 

In [402]:
SATvsGPA.isnull().sum()

SAT    0
GPA    0
dtype: int64

## Dataset 3

### Step 1: Deleting unnecessary columns

-  We only need the columns: Name of University, Rigor, Rank, GPA, Test Score 

-  We don't need the columns: Essay, Recommendation, Interview, Extracurricular, Talent, Personality, First Generation, Religious, State Residency, Race, Volunteer, Work Experience, Level of Interest

-  The column Rigor is obtained because academic rigor affects the GPA Average. For example, universities with high consideration for academic rigor might have lower regard towards applicants' GPA 

In [403]:
admission_factors = score_importance('Admission Data.csv')
admission_factors.drop(['Essay', 'Recommendation', 'Interview', 'Extracurricular', 'Talent', 'Personality', 'First Generation', 'Religious', 'State Residency', 'Race', 'Volunteer', 'Work Experience', 'Level of Interest'], axis=1, inplace=True)
admission_factors

Name of University  Rigor  Rank  GPA  Test Score
0                           Princeton University      3     3    3           3
1                               Brown University      3     3    3           3
2   California Institute of Technology (Caltech)      3     2    2           3
3                             Cornell University      3     2    3           3
4                              Dartmouth College      3     3    3           3
5                             Harvard University      1     1    1           1
6    Massachusetts Institute of Technology (MIT)      2     1    2           2
7                            Stanford University      3     3    3           3
8                     University of Pennsylvania      3     2    1           1
9     Washington University in St. Louis (WashU)      3     3    3           3
10                               Yale University      3     3    3           1
11                               Rice University      3     3    3           3
12                               Duke University      3     3    3           3
13                         Vanderbilt University      3     3    3           3
14                       Northwestern University      3     3    3           1
15       University of Southern California (USC)      3     0    3           3
16                      University of Notre Dame      3     2    2           1
17              Carnegie Mellon University (CMU)      3     3    3           2
18                         Georgetown University      3     3    3           3
19                     New York University (NYU)      3     3    3           3
20               The Ohio State University (OSU)      3     3    3           2
21                              Miami University      3     3    3           3
22         Case Western Reserve University (CWR)      3     3    3           1

### Step 2: Processing N/A values

Check if there is any N/A values using the `isnull()` function and calculate the number of N/A values spotted with `sum()`

In [404]:
admission_factors.isnull().sum()

Name of University    0
Rigor                 0
Rank                  0
GPA                   0
Test Score            0
dtype: int64

After processing the N/A values, we need to set the name of universities as the index for better data accessibility

In [405]:
admission_factors.set_index('Name of University', inplace=True)
admission_factors

Rigor  Rank  GPA  Test Score
Name of University                                                        
Princeton University                              3     3    3           3
Brown University                                  3     3    3           3
California Institute of Technology (Caltech)      3     2    2           3
Cornell University                                3     2    3           3
Dartmouth College                                 3     3    3           3
Harvard University                                1     1    1           1
Massachusetts Institute of Technology (MIT)       2     1    2           2
Stanford University                               3     3    3           3
University of Pennsylvania                        3     2    1           1
Washington University in St. Louis (WashU)        3     3    3           3
Yale University                                   3     3    3           1
Rice University                                   3     3    3           3
Duke University                                   3     3    3           3
Vanderbilt University                             3     3    3           3
Northwestern University                           3     3    3           1
University of Southern California (USC)           3     0    3           3
University of Notre Dame                          3     2    2           1
Carnegie Mellon University (CMU)                  3     3    3           2
Georgetown University                             3     3    3           3
New York University (NYU)                         3     3    3           3
The Ohio State University (OSU)                   3     3    3           2
Miami University                                  3     3    3           3
Case Western Reserve University (CWR)             3     3    3           1

## Analysis and Conclusion

### Analysis



In the first DataFrame, we can see that the top 10 universities, despite having the SAT average fluctuating around 1380 to 1450, manage to have such minimal differences in terms of applicants' GPA average of either 4 or 3.9. Furthermore, if we expand the school rankings' range to top 30, the margin for HS GPA Average increases JUST from 3.6 to 4.0 while the SAT average can go from 1160 for top schools like UCLA, 1190 for UNC Chapel Hill. This means, knowing other factors matter too, that students still have chances with a HIGH GPA and a LOW SAT 

In the second DataFrame, there is a direct correlation between individual SAT and GPA scores as students with high GPA usually have a high SAT score and vice versa. However, not all students with high GPA will have a high SAT score since the ones with 3.99 GPA have a range of SAT score from 1380 to 1520 and those with 3.91 to 3.97 GPA have a surprising range from 1207 to 1373. On the other hand, students with 2. GPAs can already score 700+ in the SAT


In the third DataFrame, there are 5 universities that consider the applicants' GPA more important than standardized test scores while there's only Caltech that prefers test scores to GPA

### Conclusion 

In conclusion, we can see a noticable gap between GPA and SAT scores in the race for college admissions. From all the analysis above, we can say that higher GPA guarantees more college acceptance compared to a high SAT score